### PR TITLE
add support for relative step sizes

### PR DESCRIPTION
**Pull request by FFroehlich**

### CODE REVIEW

File: fiddy/step.py
Comment (dweindl) [inline fiddy/step.py:46]:
```suggestion
            warnings.warn(f'point is zero valued in dimension {dimension}, '
                          f'resulting in zero step.')
```

I am wondering if this could be handled better. Provide a fall-back absolute step in case the step would be zero? Interpret the supposedly relative step as absolut? Fine as is for now.

**Review by dweindl — APPROVED** [review]
:+1:

**Comment by dweindl** [comment]
@dilpath Which of the tests are expected to pass? I think the ones expected to fail should either xfail, or be moved to a separate pull request.

**Comment by dilpath** [comment]
Thanks for the various PRs @FFroehlich! Unfortunately, there is a redesign of this package (the `redesign` branch, examples in [1]), which has various improvements, and I was waiting for feedback on it via #16 before merging. However, I didn't anticipate others submitting additional PRs in the meantime.

I intend to incorporate the work done in your PRs into `redesign`. Is there a time-critical element to this -- would you like to have these PRs merged into `main` soon? If so, I can review them now, ignoring the redesign for now.

[1] https://github.com/ICB-DCM/fiddy/blob/redesign/examples/derivative.ipynb

**Comment by dilpath** [comment]
done in #20

**Pull request closed without merging**
